Assets/Scripts/Game/MagicAndEffects/Effects/Enchanting/VampiricEffect.cs

```diff
@@ -33,7 +33,7 @@ public override void SetProperties()
             properties.GroupName = TextManager.Instance.GetText(textDatabase, EffectKey);
             properties.ShowSpellIcon = false;
             properties.AllowedCraftingStations = MagicCraftingStations.ItemMaker;
-            properties.ItemMakerFlags = ItemMakerFlags.AllowMultiplePrimaryInstances;
+            properties.ItemMakerFlags = ItemMakerFlags.AllowMultiplePrimaryInstances | ItemMakerFlags.WeaponOnly;
```

Comment:
Looking at this again, we actually should not add this flag here. It's perfectly valid for Vampiric Effect to be added to armour items using At Range variant.